lib/oli_web/components/delivery/scored_activities/scored_activities.ex

```diff
@@ -520,51 +524,50 @@ defmodule OliWeb.Components.Delivery.ScoredActivities do
     )
   end
 
-  defp get_activities(%{content: content} = _current_assessment, _section_slug)
-       when content == %{},
-       do: []
-
-  defp get_activities(current_assessment, section_slug) do
-    current_assessment.content["model"]
-    |> Enum.filter(fn element -> element["type"] == "activity-reference" end)
-    |> Enum.map(fn activity -> activity["activity_id"] end)
-    |> case do
-      [] ->
-        []
-
-      activity_ids ->
-        activities =
-          DeliveryResolver.activities_by_resource_ids(
-            activity_ids,
-            section_slug
+  defp get_activities(current_assessment, section) do
+    activities =
+      from(aa in ActivityAttempt,
+        join: res_attempt in ResourceAttempt,
+        on: aa.resource_attempt_id == res_attempt.id,
+        where: res_attempt.revision_id == ^current_assessment.id,
+        join: res_access in ResourceAccess,
+        on: res_attempt.resource_access_id == res_access.id,
+        where: res_access.section_id == ^section.id,
+        join: rev in Revision,
+        on: aa.revision_id == rev.id,
+        group_by: rev.id,
+        select: {rev, count(aa.id), sum(aa.score) / sum(aa.out_of)}
+      )
+      |> Repo.all()
+      |> Enum.map(fn {rev, total_attempts, avg_score} ->
```

Comment:
This query joins from the ActivityAttempt, through to the Revision based on the revision ID from that attempt.  It then merges in `total_attempts` and `avg_score` into that Revision.  The revisions are then used as the basis for the table model.  

This is problematic in that it incorrectly handles the case when there are more than one revision present for a single resource_id.  To replicate that situation:

1. Create an assessment with one question.  Publish the course and have one student take an attempt on that page.
2. Then, as an author, make a change to the activity (add a new choice, for instance).  Republish and have the student take a second attempt.

This UI will then incorrectly show two rows for the same activity, since this is driven from revision, and not from resource id uniqueness.

Reply:
I could not reproduce the bug you mentioned, since the current_assessment already corresponds to the latest revision (assigned in the instructor_dashboard_live [here](https://github.com/Simon-Initiative/oli-torus/blob/eff16f50cf0c711c3366166497c09a004e23c3c6/lib/oli_web/live/delivery/instructor_dashboard/instructor_dashboard_live.ex#L113)).

https://github.com/Simon-Initiative/oli-torus/assets/74839302/8812b036-4e13-48f7-b00b-a72186dd77c1

Reply:
I misunderstood your comment and finally could replicate the error. I will let you know as soon as I fix it.